### PR TITLE
Use 'crowbarctl upgrade status' to check the upgrade progress

### PR DESCRIPTION
**Pull request by jsuchome**
This is what user is expected to do.

Not tested

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4695,7 +4695,7 @@ function onadmin_prepare_crowbar_upgrade
         safely crowbar_api_request POST $crowbar_api /installer/upgrade/prepare.json
     else
         safely crowbarctl upgrade prepare
-        wait_for 300 5 "grep current_step $upgrade_progress_file | grep -v prepare" "prepare step to finish"
+        wait_for 300 10 "crowbarctl upgrade status | grep 'steps.prepare.status' | grep -q passed" "prepare step to finish"
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: would this work as well?
`... | grep -q 'steps.prepare.status.*passed'
If so, I would recommend it, as it is shorter.

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4883,12 +4883,12 @@ function onadmin_allow_vendor_change_at_nodes
 
 function crowbar_nodeupgrade_finished
 {
-    if [ ! -e /var/lib/crowbar/upgrade/6-to-7-upgrade-running ] ; then
+    if crowbarctl upgrade status | grep 'steps.nodes' | grep -q passed; then
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: same here

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -4906,15 +4906,15 @@ function onadmin_crowbar_nodeupgrade
         if [[ $want_nodesupgrade ]]; then
             safely crowbarctl upgrade services
 
-            wait_for 300 5 "grep current_step $upgrade_progress_file | grep -v services" "services step to finish"
+            wait_for 300 10 "crowbarctl upgrade status | grep 'steps.services.status' | grep -q passed" "services step to finish"
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: and here

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -4906,15 +4906,15 @@ function onadmin_crowbar_nodeupgrade
...
 
             if grep -q "failed" $upgrade_progress_file ; then
                 crowbarctl upgrade status
                 complain 12 "'Services' step has failed. Check the upgrade status."
             fi
 
             safely crowbarctl upgrade backup openstack
-            wait_for 300 5 "grep current_step $upgrade_progress_file | grep -v backup_openstack" "backup openstack step to finish"
+            wait_for 300 10 "crowbarctl upgrade status | grep 'steps.backup_openstack.status' | grep -q passed" "backup openstack step to finish"
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: and here

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: maybe introduce a small helper function for above waitfor status | grep $regexp

**Comment by jdsn** [comment]
@jsuchome is this still needed (no reply since almost 2 months)

**Comment by jsuchome** [comment]
It has lower priority, so I did not have time for it

**Comment by jdsn** [comment]
@jsuchome is this PR sill needed? (no reply for another few months)

**Comment by jsuchome** [comment]
There's no time for this ... :-(

**Comment by jdsn** [comment]
Quick proposal: https://github.com/jdsn/SUSE-Cloud--automation/commits/check-upgrade-status

**Pull request closed without merging**